File: 0729. My Calendar I.cpp
class MyCalendar
{
    vector<pair<int, int>> v;
public:
    MyCalendar()
    {
        
    }
    
    bool book(int start, int end)
    {
        for (auto p : v)
        {
            if (p.first < end and start < p.second)
            {
                return false;
            }   
        }
        
        v.push_back(make_pair(start, end));
        return true;
    }
};

/**
 * Your MyCalendar object will be instantiated and called as such:
 * MyCalendar* obj = new MyCalendar();
 * bool param_1 = obj->book(start,end);
 */

File: 0729. My Calendar I.cs
public class MyCalendar
{
    private readonly List<(int Start, int End)> _events = new ();
    
    public bool Book(int start, int end)
    {
        if (_events.Any(x => x.Start < end && x.End > start))
        {
            return false;
        }

        _events.Add((start, end));

        return true;
    }
}

/**
 * Your MyCalendar object will be instantiated and called as such:
 * MyCalendar obj = new MyCalendar();
 * bool param_1 = obj.Book(start,end);
 */